packages/web-components/fast-foundation/src/menu-item/menu-item.template.ts

```diff
@@ -30,25 +30,21 @@ export function menuItemTemplate(
             ${when(
                 x => x.role === MenuItemRole.menuitemcheckbox,
                 html<FASTMenuItem>`
-                    <div part="input-container" class="input-container">
-                        <span part="checkbox" class="checkbox">
-                            <slot name="checkbox-indicator">
-                                ${options.checkboxIndicator || ""}
-                            </slot>
-                        </span>
-                    </div>
+                    <span part="checkbox-indicator" class="checkbox-indicator">
+                        <slot name="checkbox-indicator">
+                            ${options.checkboxIndicator ?? ""}
+                        </slot>
+                    </span>
                 `
             )}
             ${when(
                 x => x.role === MenuItemRole.menuitemradio,
                 html<FASTMenuItem>`
-                    <div part="input-container" class="input-container">
-                        <span part="radio" class="radio">
-                            <slot name="radio-indicator">
-                                ${options.radioIndicator || ""}
-                            </slot>
-                        </span>
-                    </div>
+                    <span part="radio-indicator" class="radio-indicator">
+                        <slot name="radio-indicator">
+                            ${options.radioIndicator ?? ""}
+                        </slot>
+                    </span>
                 `
             )}
```

Comment:
Working on replacing Anchored Region in Menu item - not going to touch these, but I do think we need to thin out these when directives in favor of a more performant approach (same with the submenu reference below). That can be a follow-up though - wanted to note it here though as it came up. This could probably get extracted into a single template that applies the right radio/checkbox class/part/options. @EisenbergEffect as FYI

Reply:
This has been extracted to a single template.